packages/steppp/src/index.ts

```diff
@@ -99,31 +128,47 @@ function Steppp(element: HTMLElement, options: any = defaultOptions): Instance {
       afterRepaint(async () => {
         const fallbackIncrementor = direction === "forward" ? 1 : -1;
         const oldActiveStep = getStep();
+
         const newActiveStep =
           getStepByName(stepName) ||
           getStep(getActiveStepIndex() + fallbackIncrementor);
+
         const eventArgs = {
           oldStep: oldActiveStep,
           newStep: newActiveStep,
           element,
         };
 
-        if (direction === "forward" && !(await stepIsValid(getStep()))) {
+        // We're at the end of the line. Don't bother with anything else.
+        if (!newActiveStep) {
```

Comment:
I moved the order of these checks because some of the following code requires that the `newActiveStep` exists.